### PR TITLE
Powermax manila Support

### PR DESCRIPTION
**Pull request by rajinir**

### CODE REVIEW

File: src/deploy/osp_deployer/director.py [hunk 1]
```diff
@@ -1057,6 +1067,49 @@ def setup_powermax_cinder(self, powermax_cinder_yaml):
...
+                'sed -i "s|<manila_powermax_nas_login>|' +
+                self.settings.manila_powermax_nas_login + '|" ' +
+                powermax_manila_yaml,
+                'sed -i "s|<manila_powermax_nas_password>|' +
+                self.settings.manila_powermax_nas_password + '|" ' +
+                powermax_manila_yaml,
+                'sed -i "s|<manila_powermax_nas_server>|' +
+                self.settings.manila_powermax_nas_server + '|" ' +
+                powermax_manila_yaml,
+                'sed -i "s|<manila_powermax_server_container>|' +
```
Comment (gaelrehault) [inline src/deploy/osp_deployer/director.py]: Looks like a typo here , should be manila_powermax_server_container**s**

File: src/pilot/templates/powermax-manila-config.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  # Only manila-share is pacemaker managed:
+  OS::TripleO::Services::ManilaShare: ./overcloud/docker/services/pacemaker/manila-share.yaml
+  OS::TripleO::Services::ManilaBackendVMAX: ./overcloud/puppet/services/manila-backend-vmax.yaml
+
+parameter_defaults:
+  ManilaVMAXBackendName: tripleo_manila_vmax
+  ManilaVMAXDriverHandlesShareServers: <manila_powermax_driver_handles_share_servers>
+  ManilaVMAXNasLogin: <manila_powermax_nas_login>
+  ManilaVMAXNasPassword: <manila_powermax_nas_password>
+  ManilaVMAXNasServer: <manila_powermax_nas_password>
```
Comment (gaelrehault) [inline src/pilot/templates/powermax-manila-config.yaml]: Shouldn't this be manila_powermax_nas_server  ?

File: src/deploy/osp_deployer/director.py [hunk 3]
```diff
@@ -1057,6 +1067,49 @@ def setup_powermax_cinder(self, powermax_cinder_yaml):
...
 
+    def setup_powermax_manila(self, powermax_manila_yaml):
+
+        if self.settings.enable_powermax_manila_backend is False:
+            logger.debug("Not setting up powermax manila backend.")
+            return
+
+        logger.debug("Configuring dell emc powermax manila backend.")
+
+        overcloud_images_file = self.home_dir + "/overcloud_images.yaml"
```
Comment (gaelrehault) [inline src/deploy/osp_deployer/director.py]: Should this be there ?

File: src/deploy/osp_deployer/settings/sample_xsp_profile.ini [hunk 4]
```diff
@@ -525,6 +525,19 @@ powermax_array=CHANGEME
...
+#
+enable_powermax_manila_backend=false
+manila_powermax_driver_handles_share_servers=true
+manila_powermax_nas_login=CHANGEME
+manila_powermax_nas_password=CHANGEME
+manila_powermax_nas_server=CHANGEME
+manila_powermax_server_container=CHANGEME
+manila_powermax_share_data_pools=CHANGEME
+manila_powermax_ethernet_ports=CHANGEME
+
```
Comment (gaelrehault) [inline src/deploy/osp_deployer/settings/sample_xsp_profile.ini]: Probably need to update the comments section above to describe how to use this settings, like the rest of backends

File: src/deploy/osp_deployer/director.py (hunk 3 above)
Comment (rajinir) [reply src/deploy/osp_deployer/director.py]: Good catch, will remove

File: src/deploy/osp_deployer/director.py (hunk 1 above)
Comment (rajinir) [reply src/deploy/osp_deployer/director.py]: It is the name of the data mover. it is server_container

File: src/deploy/osp_deployer/settings/sample_xsp_profile.ini (hunk 4 above)
Comment (rajinir) [reply src/deploy/osp_deployer/settings/sample_xsp_profile.ini]: Yes, will do

File: src/pilot/templates/powermax-manila-config.yaml (hunk 2 above)
Comment (rajinir) [reply src/pilot/templates/powermax-manila-config.yaml]: Yes, it was a typo, will fix. Good catch

File: src/deploy/osp_deployer/settings/sample_xsp_profile.ini (hunk 4 above)
Comment (rajinir) [reply src/deploy/osp_deployer/settings/sample_xsp_profile.ini]: > Probably need to update the comments section above to describe how to use this settings, like the rest of backends

Done

File: src/deploy/osp_deployer/director.py (hunk 1 above)
Comment (gaelrehault) [reply src/deploy/osp_deployer/director.py]: it s container**b** with a "s" in the yaml file ? so the operation without the s at the end above wont sed properly

Comment (rajinir) [reply src/deploy/osp_deployer/director.py]: You are right, I fixed the yaml file

**Comment by rajinir** [comment]
Tested the deployment, it was successful. To do functional testing have to wait for the File Share to be configured in the backend. This PR should be ready to go now

**Comment by heydog15** [comment]
Rajini fixed all of Gael's concerns.  Going to merge.

**Pull request merged**